sycl/include/sycl/ext/oneapi/experimental/annotated_ptr/annotated_ptr.hpp

```diff
@@ -91,15 +62,44 @@ class annotated_ref<T, detail::properties_t<Props...>> {
   T *m_Ptr;
   explicit annotated_ref(T *Ptr) : m_Ptr(Ptr) {}
 
+  // properties filter
+  template <typename property_list, template <class...> typename filter>
+  using PropertiesFilter =
+      sycl::detail::boost::mp11::mp_copy_if<property_list, filter>;
+
+  template <typename p>
+  using annotation_filter = propagateToPtrAnnotation<typename p::key_t>;
+
+  // filter properties that are applied on annotations
+  using annotation_props = sycl::detail::boost::mp11::mp_rename<
+      PropertiesFilter<property_list_t, annotation_filter>,
+      detail::properties_t>;
```

Comment:
what are you calling rename for? This is already properties_t<...> after your other change.

```suggestion
  using annotation_props = PropertiesFilter<property_list_t, annotation_filter>;
```